--- a/src/ui/wxWidgets/PasswordSafeFrame.h	2025-11-22 04:17:08.000000000 -0500
+++ b/src/ui/wxWidgets/PasswordSafeFrame.h	2026-04-21 10:24:13.124614503 -0400
@@ -42,6 +42,9 @@
 #include <utility>
 #include <vector>
 
+// Custom event fired when GUI-related DB prefs change.
+wxDECLARE_EVENT(wxEVT_GUI_DB_PREFS_CHANGE, wxCommandEvent);
+
 /*!
  * Forward declarations
  */
@@ -223,7 +226,6 @@
 class PasswordSafeFrame : public wxFrame, public Observer, public wxModalDialogHook
 {
     DECLARE_CLASS( PasswordSafeFrame )
-    DECLARE_EVENT_TABLE()
 
 private:
     enum class ViewType { TREE, GRID };
@@ -245,6 +247,7 @@
   /// Creates the controls and sizers
   void CreateControls();
   void CreateMenubar();
+  void BindEvents();
 
   ItemList::size_type GetNumEntries() const {return m_core.GetNumEntries();}
 
@@ -890,8 +893,4 @@
   wxTopLevelWindow* m_pengingCloseWindow = nullptr; // current window that processing close event
 };
 
-BEGIN_DECLARE_EVENT_TYPES()
-DECLARE_EVENT_TYPE(wxEVT_GUI_DB_PREFS_CHANGE, -1)
-END_DECLARE_EVENT_TYPES()
-
 #endif // _PASSWORDSAFEFRAME_H_
