--- a/src/ui/wxWidgets/ManagePasswordPoliciesDlg.cpp	2025-11-22 04:17:08.000000000 -0500
+++ b/src/ui/wxWidgets/ManagePasswordPoliciesDlg.cpp	2026-04-23 12:48:07.927060545 -0400
@@ -43,46 +43,6 @@
 IMPLEMENT_CLASS( ManagePasswordPoliciesDlg, wxDialog )
 
 /*!
- * ManagePasswordPoliciesDlg event table definition
- */
-
-BEGIN_EVENT_TABLE( ManagePasswordPoliciesDlg, wxDialog )
-
-////@begin ManagePasswordPoliciesDlg event table entries
-  EVT_GRID_SELECT_CELL( ManagePasswordPoliciesDlg::OnSelectCell )
-
-  EVT_BUTTON( wxID_NEW, ManagePasswordPoliciesDlg::OnNewClick )
-
-  EVT_BUTTON( ID_EDIT_PP, ManagePasswordPoliciesDlg::OnEditClick )
-
-  EVT_BUTTON( wxID_DELETE, ManagePasswordPoliciesDlg::OnDeleteClick )
-
-  EVT_BUTTON( ID_LIST, ManagePasswordPoliciesDlg::OnListClick )
-
-  EVT_BUTTON( wxID_UNDO, ManagePasswordPoliciesDlg::OnUndoClick )
-
-  EVT_BUTTON( wxID_REDO, ManagePasswordPoliciesDlg::OnRedoClick )
-
-  EVT_BUTTON( ID_GENERATE_PASSWORD, ManagePasswordPoliciesDlg::OnGeneratePasswordClick )
-
-  EVT_BUTTON( ID_BITMAPBUTTON, ManagePasswordPoliciesDlg::OnCopyPasswordClick )
-
-  EVT_BUTTON( wxID_OK, ManagePasswordPoliciesDlg::OnOkClick )
-
-  EVT_BUTTON( wxID_CANCEL, ManagePasswordPoliciesDlg::OnCancelClick )
-
-  EVT_BUTTON( wxID_HELP, ManagePasswordPoliciesDlg::OnHelpClick )
-
-////@end ManagePasswordPoliciesDlg event table entries
-
-  EVT_SIZE( ManagePasswordPoliciesDlg::OnSize )
-
-  EVT_MAXIMIZE( ManagePasswordPoliciesDlg::OnMaximize )
-
-  EVT_CLOSE( ManagePasswordPoliciesDlg::OnClose )
-END_EVENT_TABLE()
-
-/*!
  * ManagePasswordPoliciesDlg constructor
  */
 
@@ -102,6 +62,24 @@
   SetExtraStyle(wxWS_EX_BLOCK_EVENTS);
   wxDialog::Create( parent, id, caption, pos, size, style );
 
+  // TODO: for EVT_GRID_SELECT_CELL:   EVT_GRID_SELECT_CELL( ManagePasswordPoliciesDlg::OnSelectCell )
+  Bind(wxEVT_GRID_SELECT_CELL, &ManagePasswordPoliciesDlg::OnSelectCell, this);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnNewClick, this, wxID_NEW);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnEditClick, this, ID_EDIT_PP);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnDeleteClick, this, wxID_DELETE);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnListClick, this, ID_LIST);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnUndoClick, this, wxID_UNDO);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnRedoClick, this, wxID_REDO);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnGeneratePasswordClick, this, ID_GENERATE_PASSWORD);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnCopyPasswordClick, this, ID_BITMAPBUTTON);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnOkClick, this, wxID_OK);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnCancelClick, this, wxID_CANCEL);
+  Bind(wxEVT_BUTTON, &ManagePasswordPoliciesDlg::OnHelpClick, this, wxID_HELP);
+  Bind(wxEVT_SIZE, &ManagePasswordPoliciesDlg::OnSize, this);
+  // TODO: unexpected args for EVT_MAXIMIZE:   EVT_MAXIMIZE( ManagePasswordPoliciesDlg::OnMaximize )
+  Bind(wxEVT_MAXIMIZE, &ManagePasswordPoliciesDlg::OnMaximize, this);
+  Bind(wxEVT_CLOSE_WINDOW, &ManagePasswordPoliciesDlg::OnClose, this);
+
   CreateControls();
   if (GetSizer())
   {
