--- a/src/ui/wxWidgets/PasswordSubsetDlg.cpp	2025-11-22 04:17:08.000000000 -0500
+++ b/src/ui/wxWidgets/PasswordSubsetDlg.cpp	2026-04-28 21:16:09.824723080 -0400
@@ -38,20 +38,6 @@
 IMPLEMENT_DYNAMIC_CLASS( PasswordSubsetDlg, wxDialog )
 
 /*!
- * PasswordSubsetDlg event table definition
- */
-
-BEGIN_EVENT_TABLE( PasswordSubsetDlg, wxDialog )
-
-////@begin PasswordSubsetDlg event table entries
-  EVT_BUTTON( ID_BITMAPBUTTON, PasswordSubsetDlg::OnBitmapbuttonClick )
-  EVT_BUTTON( wxID_CLOSE, PasswordSubsetDlg::OnCloseClick )
-  EVT_TEXT( ID_TEXTCTRL_POS, PasswordSubsetDlg::OnPosListChanged )
-////@end PasswordSubsetDlg event table entries
-
-END_EVENT_TABLE()
-
-/*!
  * PasswordSubsetDlg constructors
  */
 PasswordSubsetDlg::PasswordSubsetDlg(wxWindow *parent, const StringX &password,
@@ -63,6 +49,10 @@
   SetExtraStyle(wxWS_EX_VALIDATE_RECURSIVELY|wxWS_EX_BLOCK_EVENTS);
   wxDialog::Create( parent, id, caption, pos, size, style );
 
+  Bind(wxEVT_BUTTON, &PasswordSubsetDlg::OnBitmapbuttonClick, this, ID_BITMAPBUTTON);
+  Bind(wxEVT_BUTTON, &PasswordSubsetDlg::OnCloseClick, this, wxID_CLOSE);
+  Bind(wxEVT_TEXT, &PasswordSubsetDlg::OnPosListChanged, this, ID_TEXTCTRL_POS);
+
   CreateControls();
   if (GetSizer())
   {
