--- a/src/ui/wxWidgets/PWFiltersPasswordDlg.cpp	2025-11-22 04:17:08.000000000 -0500
+++ b/src/ui/wxWidgets/PWFiltersPasswordDlg.cpp	2026-05-03 18:01:15.051988435 -0400
@@ -50,21 +50,6 @@
 IMPLEMENT_CLASS( pwFiltersPasswordDlg, wxDialog )
 
 /*!
- * pwFiltersPasswordDlg event table definition
- */
-
-BEGIN_EVENT_TABLE( pwFiltersPasswordDlg, wxDialog )
-
-  EVT_BUTTON( wxID_OK, pwFiltersPasswordDlg::OnOk )
-  EVT_COMBOBOX( ID_COMBOBOX72, pwFiltersPasswordDlg::OnSelectionChange )
-  EVT_TEXT( ID_TEXTCTRL73, pwFiltersPasswordDlg::OnTextChange )
-  EVT_SPINCTRL( ID_SPINCTRL75, pwFiltersPasswordDlg::OnFNum1Change )
-  EVT_BUTTON( wxID_CANCEL, pwFiltersPasswordDlg::OnCancelClick )
-  EVT_CLOSE( pwFiltersPasswordDlg::OnClose )
-
-END_EVENT_TABLE()
-
-/*!
  * pwFiltersPasswordDlg constructors
  */
 pwFiltersPasswordDlg::pwFiltersPasswordDlg(wxWindow *parent, FieldType ftype, PWSMatch::MatchRule *rule, wxString *value, bool *fcase, int *fnum1)
@@ -76,7 +61,14 @@
   Init();
 
   SetExtraStyle(wxWS_EX_BLOCK_EVENTS);
-  wxDialog::Create(parent, wxID_ANY, _("Display Filter Password Value"), wxDefaultPosition, wxDefaultSize, wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER);
+  wxDialog::Create(parent, wxID_ANY, wxTRANSLATE("Display Filter Password Value"), wxDefaultPosition, wxDefaultSize, wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER);
+
+  Bind(wxEVT_BUTTON, &pwFiltersPasswordDlg::OnOk, this, wxID_OK);
+  Bind(wxEVT_COMBOBOX, &pwFiltersPasswordDlg::OnSelectionChange, this, ID_COMBOBOX72);
+  Bind(wxEVT_TEXT, &pwFiltersPasswordDlg::OnTextChange, this, ID_TEXTCTRL73);
+  Bind(wxEVT_SPINCTRL, &pwFiltersPasswordDlg::OnFNum1Change, this, ID_SPINCTRL75);
+  Bind(wxEVT_BUTTON, &pwFiltersPasswordDlg::OnCancelClick, this, wxID_CANCEL);
+  Bind(wxEVT_CLOSE_WINDOW, &pwFiltersPasswordDlg::OnClose, this);
 
   CreateControls();
 
@@ -171,7 +163,7 @@
   wxTimeSpan elapsedTime = wxDateTime(18, wxDateTime::Jan, 2038) - wxDateTime::Now();
   m_max = elapsedTime.GetDays();
   
-  infoFmtStr = _("Interval [%d to %d]");
+  infoFmtStr = wxTRANSLATE("Interval [%d to %d]");
 }
 
 /*!
@@ -367,7 +359,7 @@
     if(*m_prule != PWSMatch::MR_EXPIRED &&
        *m_prule != PWSMatch::MR_WILLEXPIRE &&
        m_string.IsEmpty()) {
-      wxMessageBox(_("Specify text."), _("Missing text for the selected rule."), wxOK|wxICON_ERROR);
+      wxMessageBox(wxTRANSLATE("Specify text."), wxTRANSLATE("Missing text for the selected rule."), wxOK|wxICON_ERROR);
       return;
     }
     
